src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -861,6 +1081,100 @@ JoystickCaps JoystickImpl::getCapabilitiesDInput() const
     return caps;
 }
 
+////////////////////////////////////////////////////////////
+JoystickState JoystickImpl::updateXInput()
+{
+    // No identified index yet
+    if (m_xInputIndex == 0xFFFFFFFF)
+    {
+        for (std::size_t i = 0; i < xInputSlots.size(); ++i)
+        {
+            const bool taken = xInputSlots[i];
+            if (!taken)
+            {
+                xInputSlots[i] = true;
+                m_xInputIndex  = static_cast<DWORD>(i);
+                break;
+            }
+        }
+    }
+
+    XINPUT_STATE xinputState{};
+    const auto   result = mXInputGetState(m_xInputIndex, &xinputState);
+
+    if (result != S_OK)
+    {
+        // probably device not connected.
+        xInputSlots[m_xInputIndex] = false;
+        m_state                    = {};
+        m_xInputIndex              = 0xFFFFFFFF;
+        return m_state;
+    }
+
+    // INFO: After consideration, the Directional Pad will be exposed as PovX and PovY axes for consistency with PS5 DualSense controllers.
+
+    auto& state      = m_state;
+    auto& gamepad    = xinputState.Gamepad;
+    state.buttons[0] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_A);
+    state.buttons[1] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_B);
+    state.buttons[2] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_X);
+    state.buttons[3] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_Y);
+    state.buttons[4] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_START);
+    state.buttons[5] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_BACK);
+    state.buttons[6] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_LEFT_SHOULDER);
+    state.buttons[7] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_RIGHT_SHOULDER);
+    state.buttons[8] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_LEFT_THUMB);
+    state.buttons[9] = 0 != (gamepad.wButtons & XINPUT_GAMEPAD_RIGHT_THUMB);
+
+    // The standard deadzone felt, too... dead. This feels reasonable, but should it be configurable?
+    // Threshold is ... different, and NOT a deadzone, but instead a required delta between inputs to event an update.
+    static constexpr SHORT deadzone = XINPUT_GAMEPAD_LEFT_THUMB_DEADZONE / 2;
```

Comment:
Please note that there is also `XINPUT_GAMEPAD_RIGHT_THUMB_DEADZONE`

Reply:
The XInput sample from the DirectX SDK uses another dead zone:
https://github.com/microsoft/DirectX-SDK-Samples/blob/main/C%2B%2B/XInput/SimpleController/SimpleController.cpp#L39
which translates to `7864`
That is slightly bigger than the `7849` from `XINPUT_GAMEPAD_LEFT_THUMB_DEADZONE`